src/components/App.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import CryptoFinder from "./CryptoFinder";
+import SavedCryptos from "./SavedCryptos";
+import "./App.css";
+import SavedCrypto from "../models/SavedCrypto";
+import Header from "./Header";
+
+const queryClient = new QueryClient();
+
+const App: React.FC = () => {
+  const [favCryptos, setFavCryptos] = useState<SavedCrypto[]>(() => {
+    const saved = localStorage.getItem("savedCoins");
+    return saved ? JSON.parse(saved) : [];
+  });
+
+  function handleFavCrypto(selectedCrypto: SavedCrypto) {
```

Comment:
The caller of this function should specify whether the coin is being added to favorites or removed. This would help you avoid a few checks

Reply:
I have added a new parameter to this function - isSaved, which basically replaces the exists variable.